### PR TITLE
fix(KMS-19979): Showing hint saves answer after edit for open-question

### PR DESCRIPTION
**Pull request by shalommeoded**
in case where the user want to change his answer after he type the new text then open hint and close it the new text is saved automatically.

Added a logic, when the isAnswered is set to true, we display the edit mode.

P.S, the answer will get overridden only on the display not on the answer it self.

### CODE REVIEW

File: modules/Quiz/resources/quiz.js [hunk 1]
```diff
@@ -559,9 +559,18 @@
...
-                        if (holdAnswer.length !== 0) {
+                        if (!cPo.isAnswerd) {
+                            interfaceElement.find(".open-question-textarea").val(holdAnswer);
+                            interfaceElement.find(".open-question-chars .chars").text(holdAnswer.length);
+                            if (holdAnswer.length !== 0) {
+                                interfaceElement.find("#open-question-clear,#open-question-save").removeAttr("disabled");
+                            }
+                        } else {
+                            interfaceElement.find(".open-question-textarea").val(holdAnswer);
+                            interfaceElement.find(".open-question-chars .chars").text(holdAnswer.length);
```
Comment (semarche-kaltura) [inline modules/Quiz/resources/quiz.js]: same method on 563 and 569, on 564 and 570.
we can move them to own function, like
```
function setAnswer(answer) {
     interfaceElement.find(".open-question-textarea").val(answer);
     interfaceElement.find(".open-question-chars .chars").text(answer.length);
}
```

**Comment by shalommeoded** [comment]
@eitanavgil  @semarche-kaltura 
Done, can we merge this?

File: modules/Quiz/resources/quiz.js [hunk 2]
```diff
@@ -559,12 +559,19 @@
                 .click(function () {
                     $(".close-button")
                     .click(function () {
-                        interfaceElement.find(".open-question-textarea").val(holdAnswer);
-                        interfaceElement.find(".open-question-chars .chars").text(holdAnswer.length);
-                        if (holdAnswer.length !== 0) {
+                        if (!cPo.isAnswerd) {
```
Comment (eitanavgil) [inline modules/Quiz/resources/quiz.js]: @shalommeoded  this is wrong ! 
Try the next scenario: 
Make an open-question with hint quiz 
answer the quiz with answer "11111111" and save your answer 
submit the quiz and replay 
see question again (after submit) and click "hint"  
Wrongly see edit-mode

Comment (shalommeoded) [reply modules/Quiz/resources/quiz.js]: Fixed

**Comment by OrenMe** [comment]
what is the status of this PR?

**Pull request closed without merging**